packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +273,526 @@ class Cubic extends Curve {
   }
 }
 
+/// Abstract class that defines an API for evaluating 2D parametric curves.
+abstract class Curve2D {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const Curve2D();
+
+  /// Returns the position of the curve at point `t`.
+  ///
+  /// This function must ensure that the value of `t` is between 0.0 and
+  /// 1.0, inclusive.
+  ///
+  /// It is recommended that subclasses override [transformInternal] instead of
+  /// this function, as the above case is already enforced in the default
+  /// implementation of [transform], which delegates the remaining logic to
+  /// [transformInternal].
+  Offset transform(double t) {
+    assert(t >= 0.0 && t <= 1.0, 'parameter t is ${t.toStringAsFixed(2)}, which is outside of the range [0.0, 1.0]');
+    return transformInternal(t);
+  }
+
+  /// Returns the position of the curve at point `t`, where `t` is between 0.0
+  /// and 1.0, inclusive.
+  @protected
+  Offset transformInternal(double t) {
+    throw UnimplementedError();
+  }
+
+  @override
+  String toString() => '$runtimeType';
```

Comment:
runtimeType.toString is _very_ slow.

Reply:
Yeah, but I'm not sure what else I would want to print here, and overriding it in each class to a string literal is error prone. It's generally only ever used in debug mode.

Isn't it just as slow to not override `toString` and have it print "Instance of MyType"?